content/en/docs/architecture/ci-operator.md

```diff
@@ -221,6 +221,44 @@ By making use of the previously compiled artifacts in the intermediate `pipeline
 cache the Go build. If multiple output `images` exist that rely on a previously built artifact, this caching effect can
 reduce build times dramatically.
 
+## Build inputs
+[Building Container Images](#building-container-images  ) paragraph has shown how it is possible to leverage build inputs `.inputs[]` to speed the building process up. We are going to analyze this feature in a greater detail now.
+### `as:` substitutions
+It is a list of image names that directly maps to [`.spec.source.images[].as`](https://docs.openshift.com/container-platform/4.12/rest_api/workloads_apis/build-build-openshift-io-v1.html#spec-source-images-2) of an [Openshift Build](https://docs.openshift.com/container-platform/4.12/rest_api/workloads_apis/build-build-openshift-io-v1.html#specification), therefore it preserves the semantic.  
+When the aforementioned stanza exists, a build searches for every source images that matches one of the `.as` names and it performs a substitution with the image specified in `.inputs[]`.  
+
+Images substitution can happen in one of the following ways:  
+- in a `--from=` argument:
+```yaml
+images:
+- from: src
+  dockerfile_literal: |
+    FROM itdoesntmatter # This is going be replaced by "pipeline:src"
```

Comment:
Maybe leave these out (and also the corresponding `from: src` stanzas) for clarity, to make the examples illustrate just the one mechanism (`inputs.image.as`) instead of two?

Reply:
Sounds fine.